### ISSUE TITLE
Async archiving with pg_receivexlog

### ISSUE THREAD

**Issue by dwsteele** [pgmasters/backrest#124]
Make async archiving work when pg_receivexlog is writing directly to the out spool directory on the backup server.  Settting this up with a replication slot will make it more reliable.

BackRest will not directly call pg_receivexlog but it can work in concert for more reliable logging.

Issues to look at:
1) Does pg_receivelog generate a temp file name (how to know when it is done)?
2) Can it run as a daemon?

**Comment by dwsteele** [comment]
Moved to xml backlog.

**Comment by anse1** [comment]
Hi, just wanted to inquire whether support for pg_receivexlog is still on the agenda.
Were the "issues to look at" looked at before closing of the github issue?

**Comment by dwsteele** [comment]
Several of the issues that made this less practical have been addressed in more recent versions of PostgreSQL.

While we still have an intention to support this feature (and it is is the backlog) I don't see it happening  soon since we are porting pgBackRest from Perl to C and are loathe to add any features to Perl unless we have to.  All the archive code should be ported in the next few months, though.

When we implement this it will talk directly to Postgres using the replication protocol and cut out pg_receivexlog entirely.  We may also bypass the replication protocol so we can do compression, parallelism, etc., although it would still work over a connection to Postgres.

TL;DR we are still planning a feature like this, but have not scheduled it yet.
